vocab/templates/vocab.html

```diff
@@ -73,32 +73,57 @@ <h1>Words</h1>
          <br />
          <div class="row">
             <div class="col-xs-6 text-center">
-               <form id="entry" method="POST" action="/_check">
+		    <form id="entry"> 
                   <label for="attempt">Word:</label>
                   <input id="attempt" name="attempt" type="text" width="15"
                      autocomplete="off"  autofocus />
-                  <input type="submit" name="try" />
                </form>
             </div>
          </div>
          <br />
-         {% if session.matches|length is not equalto 0  %}
+     
          <h2>You found </h2>
          <p id="results">
-            {% for match in session.matches %}
-            {{ match }}
-            {% endfor %}
-         </p>
-         {% endif %}
-         {% with messages = get_flashed_messages() %}
-         {% if messages %}
-         <!-- Mismatch, already found, etc. "flashed" from Python -->
-         {% for message in messages %}
-         <p class="message">{{ message }}</p>
-         {% endfor %}
-         {% endif %}
-         {% endwith %}
-      </div>
-      <!-- container -->
-   </body>
+	  </p>
+         <p class="message"></p>
+
+         <!-- Some logic was taken away from minijax.html -->
+<script>
+    $("#entry").submit(function(event){ 
+    // Prevent normal form submission
+    event.preventDefault();
+});
+
+let counter = 0; // Counter variable for comparing
+
+$("#attempt").keyup(function(event) {
+    var txt = $("#attempt").val(); // Current content of the input field
+    var keycode = event.which; // The key that just went up
+    var letter = String.fromCharCode(keycode);
+
+    if (! "ABCDEFGHIJKLMNOPQRSTUVWXYZ".includes(letter)) { 
+        return;
+    }
+
+    $.getJSON("/_check", { text: txt }, function(data) {
+        rsval = data.result.valid_word; 
+        console.log("rsval: " + rsval); 
+        if (rsval) {
+            counter++; 
+            $("#results").append(txt + " "); // Add word to found texts
+            $("#attempt").val(""); // Clear input field
+        }
+        console.log("sval: " + data.success.success_count); 
+        console.log("counter: " + counter); 
+        if (counter == data.success.success_count) {
+            location.replace('/completed'); // Redirect to success page if words found reaches the counter
+        }
```

Comment:
With the suggestions made to `flask_vocab.py` I'd like to make some suggestions to this code as well so messages show up in the HTML when a word is incorrect or not in the vocab. This suggestion also takes into account that `data.success` is either a 0 or a 1 

```suggestion
    $.getJSON("/_check", { text: txt }, function(data) {
        rsval = data.result.valid_word; 
        console.log("rsval: " + rsval); 
        if (rsval) {
            $("#results").append(txt + " "); // Add word to found texts
            $("#attempt").val(""); // Clear input field
            $("#error").html("");
        }
        else {
            $("#error").html(data.result.message)
        }
        
        if (data.success == 1) {
            location.replace('/completed'); // Redirect to success page if words found reaches the counter
        }
```

You could also add another else if statement that checks if the word is in the list of words but not in the anagram so the input is reset to "" to make it easier for the user to input words. This would require changing your flask implementation though and how you send back information.